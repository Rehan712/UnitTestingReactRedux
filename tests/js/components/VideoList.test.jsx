import React from 'react';
import { VideoList } from '../../../src/js/components/VideoList';

//Below are related to enzyme
import Enzyme from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import { shallow } from 'enzyme';
Enzyme.configure({ adapter: new Adapter() });

describe('VideoList Component', () => {
  let videoList, selectVideo;

  beforeEach(() => {
    videoList = [{ id: { videoId: '1' } }, { id: { videoId: '2' } }];
    selectVideo = jest.fn();
  });

  it('renders', () => {
    let wrapper = shallow(
      <VideoList
        selectVideo={selectVideo}
        videoList={videoList}
        isLoading={false}
        error={null}
      />
    );
    expect(wrapper).toMatchSnapshot();
  });

  it('renders Loading message', () => {
    let wrapper = shallow(
      <VideoList
        selectVideo={selectVideo}
        videoList={videoList}
        isLoading={true}
        error={null}
      />
    );
    expect(wrapper).toMatchSnapshot();
  });

  it('renders Error message', () => {
    let wrapper = shallow(
      <VideoList
        selectVideo={selectVideo}
        videoList={videoList}
        isLoading={false}
        error={{ message: 'Error While loading Videos' }}
      />
    );
    expect(wrapper).toMatchSnapshot();
  });

  it('passes on the selectVideo fn', () => {
    expect(selectVideo.mock.calls.length).toBe(0);
  });
});
