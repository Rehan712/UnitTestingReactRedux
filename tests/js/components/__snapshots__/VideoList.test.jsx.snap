// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VideoList Component renders 1`] = `
<div
  className="videoList"
>
  <VideoListEntry
    handleSelectedVideo={[MockFunction]}
    key="1"
    video={
      Object {
        "id": Object {
          "videoId": "1",
        },
      }
    }
  />
  <VideoListEntry
    handleSelectedVideo={[MockFunction]}
    key="2"
    video={
      Object {
        "id": Object {
          "videoId": "2",
        },
      }
    }
  />
</div>
`;

exports[`VideoList Component renders Error message 1`] = `
<div
  className="videoList"
>
  <div>
    Error While loading Videos
  </div>
</div>
`;

exports[`VideoList Component renders Loading message 1`] = `
<div
  className="videoList"
>
  <div>
    Loading Videos.....
  </div>
</div>
`;
